### ISSUE TITLE
select-next includes non-highlighted word(s)

### ISSUE THREAD

**Issue by Saucy** [atom/find-and-replace#723]
Hello, I'm having some troubles using CTRL-D to edit my next selected word, seen here:

![atom-find-next](https://cloud.githubusercontent.com/assets/2084555/15322832/40aa7994-1c3f-11e6-9d21-a72ef19e7e27.gif)

How can I ignore `site.excludes` (it doesn't even highlight it for editing)? Is there a setting somewhere in Atom I should change for this behavior to change?

- OS: Win7 64bit
- Atom version: 1.7.3
- find-and-replace version: 0.199.0

**Comment by Yajo** [comment]
I think your highlighting is from https://atom.io/packages/quick-highlight, right?

**Comment by Saucy** [comment]
No I'm using https://atom.io/packages/highlight-selected.

**Comment by Yajo** [comment]
As you can see, that package highlight some things, and this one uses others. It's normal when you use different packages. I don't think this is a bug, or at least it is not correctly explained.

**Comment by 50Wliu** [comment]
Closing as this is a conflict between packages that are using different selection behavior.

**Comment by Saucy** [comment]
What I'm using for highlight-package is not the problem!

Same gif without the highlight, still showing my problem:
![atom-find-next-02](https://cloud.githubusercontent.com/assets/2084555/15501687/1968f994-21af-11e6-95e7-be93e3ae1369.gif)

The select-next should NOT include `site.excludes`, only `exclude` (please notice the difference between the words **exclude** and **excludes**) when I use the command `find-and-replace:select-next`. Can I change this behavior?

**Comment by 50Wliu** [comment]
@Saucy how are you selecting the first `exclude`?

**Comment by Saucy** [comment]
Double-left-clicking the word, then I use my shortcut (ctrl-d) for the `find-and-replace:select-next` command that select the two others.

Sorry for very late reply. Overwatch has had my attention...

**Comment by 50Wliu** [comment]
@colllin is this expected behavior?

**Comment by colllin** [comment]
Yes, this looks like the expected behavior with the current design.  

First the text 'exclude' is selected using the mouse. Then cmd-d is pressed to trigger the SelectNext action. Even though the current selection is a whole word, it wasn't created using SelectWord, so it's treated as a character selection. SelectNext finds and selects subsequent matches as if the 'whole word' option is disabled. 

If you want SelectNext to find and select matches as if the 'whole word' option is enabled, you need to make the initial selection using the SelectWord action. Instead of double-clicking the word, click once, then press cmd-d to trigger SelectWord. Then subsequent presses of cmd-d (SelectNext) will select whole words.

**Comment by 50Wliu** [comment]
Shouldn't double-clicking a word also technically be counted as a "whole word" selection?

**Comment by colllin** [comment]
Seems reasonable to me. Is there a reliable way to detect that? Double-click event on a word?

Is there an assumption here that double-click selects the word? Should we enforce that by overriding the native behavior and calling SelectWord?

**Comment by 50Wliu** [comment]
I'm not sure if there's any events for that.  What do you mean by "is there an assumption"?

**Comment by colllin** [comment]
I just mean that it's native (osx, Windows) behavior to select a word when you double-click, isn't it? As opposed to being Atom functionality. If the native behavior changed, or someone had different settings, it could create a weird edge case unless we enforce the double-click = SelectWord behavior within Atom. Minor detail I guess.

**Comment by 50Wliu** [comment]
Yeah, it's definitely native (though I can't say if Atom has any special handling for double-clicks, since I know next-to-nothing about Atom's internals).
